File: backend/src/main/java/com/example/backend/dto/VehicleDeploymentPlanningInputDTO.java
package com.example.backend.dto;

import java.util.Set;

/**
 * Data Transfer Object for vehicle deployment planning input.
 */
public record VehicleDeploymentPlanningInputDTO (String name, Set<PersonInputDTO> persons,
                                                 Set<VehicleInputDTO> vehicles, Set<AddressInputDTO> addresses) {
}

File: frontend/src/app/dtos/VehicleDeploymentPlanningInput.dto.ts
import {PersonInputDto} from "./PersonInput.dto";
import {VehicleInputDto} from "./VehicleInput.dto";
import {AddressInputDto} from "./AddressInput.dto";

export class VehicleDeploymentPlanningInputDto {
    name: string | undefined;
    persons: PersonInputDto[] = [];
    vehicles: VehicleInputDto[] = [];
    addresses: AddressInputDto[] = [];
}